src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -430,52 +434,72 @@ public void Initialize()
 
                 var coreBinder = bindingProviders.OfType<CoreExtensionsScriptBindingProvider>().First();
                 coreBinder.AppDirectory = ScriptConfig.RootScriptPath;
 
                 // Allow BindingProviders to initialize
-                foreach (var bindingProvider in ScriptConfig.BindingProviders)
+                using (metricsLogger.LatencyEvent(MetricEventNames.HostStartupInitializeBindingProvidersLatency))
                 {
-                    try
+                    foreach (var bindingProvider in ScriptConfig.BindingProviders)
                     {
-                        bindingProvider.Initialize();
-                    }
-                    catch (Exception ex)
-                    {
-                        // If we're unable to initialize a binding provider for any reason, log the error
-                        // and continue
-                        string errorMsg = string.Format("Error initializing binding provider '{0}'", bindingProvider.GetType().FullName);
-                        TraceWriter.Error(errorMsg, ex);
-                        _startupLogger?.LogError(0, ex, errorMsg);
+                        try
+                        {
+                            bindingProvider.Initialize();
+                        }
+                        catch (Exception ex)
+                        {
+                            // If we're unable to initialize a binding provider for any reason, log the error
+                            // and continue
+                            string errorMsg = string.Format("Error initializing binding provider '{0}'", bindingProvider.GetType().FullName);
+                            TraceWriter.Error(errorMsg, ex);
+                            _startupLogger?.LogError(0, ex, errorMsg);
+                        }
                     }
+                    TraceWriter.Verbose("Binding providers initialized.");
                 }
+
                 extensionLoader.LoadBuiltinExtensions(usedBindingTypes);
 
                 var directTypes = GetDirectTypes(functionMetadata);
                 extensionLoader.LoadDirectlyReferencedExtensions(directTypes);
                 extensionLoader.LoadCustomExtensions();
+                TraceWriter.Verbose("Extension loading complete.");
 
                 // Now all extensions have been loaded, the metadata is finalized.
                 // There's a single script binding instance that services all extensions.
                 // give that script binding the metadata for all loaded extensions so it can dispatch to them.
-                var generalProvider = ScriptConfig.BindingProviders.OfType<GeneralScriptBindingProvider>().First();
-                var metadataProvider = this.CreateMetadataProvider();
-                generalProvider.CompleteInitialization(metadataProvider);
+                using (metricsLogger.LatencyEvent(MetricEventNames.HostStartupCreateMetadataProviderLatency))
```

Comment:
The operations CreateMetadataProvider/CompleteInitialization seem to just be one getters/setters. Why do you think we need to measure this?

Reply:
CreateMetadataProvider takes ~1700ms in my test.